include/zephyr/net/ieee802154_radio.h

```diff
@@ -1,12 +1,13 @@
 /*
```

Comment:
I've noticed macros in this file don't have proper Doxygen docstrings. Could you maybe have a look? They should have proper doc or be marked internal.
In fact there quite a few other places where doxygen doc is missing, but I shouldn't be hijacking this PR :)

https://docs.zephyrproject.org/latest/doxygen/html/ieee802154__radio_8h.html

Reply:
@kartben Oh, yes, you are so right - and the macros are probably the easiest problem to solve, the whole structure of the API doc doesn't make sense - much too monolithic. I was afraid to look onto that page for a long time because I knew that it means a lot of work. :-D I did so yesterday and realized how much is still missing. I worked the whole day yesterday on improving that page (PR coming soon). But I'll have to take it in small steps as I also need to progress on TSCH itself in parallel. Expect this to improve soon and steadily over the upcoming PRs, though.